packages/analytics/amplify_analytics_pinpoint_dart/lib/src/analytics_plugin_impl.dart

```diff
@@ -0,0 +1,159 @@
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'dart:async';
+
+import 'package:amplify_analytics_pinpoint_dart/amplify_analytics_pinpoint_dart.dart';
+import 'package:amplify_analytics_pinpoint_dart/src/impl/analytics_client/analytics_client.dart';
+import 'package:amplify_analytics_pinpoint_dart/src/impl/analytics_client/key_value_store.dart';
+import 'package:amplify_analytics_pinpoint_dart/src/sdk/pinpoint.dart';
+import 'package:amplify_core/amplify_core.dart';
+import 'package:amplify_secure_storage_dart/amplify_secure_storage_dart.dart';
+import 'package:aws_signature_v4/aws_signature_v4.dart';
+
+/// Public facing Plugin
+/// Validate and parse inputs
+/// Receive and provide external Flutter Provider implementations
+/// Delegate work to PinpointClient
```

Comment:
`AmplifyAnalyticsPinpoint` is the public facing package. This is also technically public, but AmplifyAnalyticsPinpoint is what will be used by customers.

```suggestion
/// The Dart implementation of AmplifyAnalyticsPinpoint.
```